x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.test.ts

```diff
@@ -330,7 +332,70 @@ describe('Host Isolation', () => {
   });
 
   describe('Cases', () => {
-    it.todo('logs a comment to the provided case');
-    it.todo('logs a comment to any cases associated with the given alert');
+    let casesClient: CasesClientMock;
+
+    beforeEach(async () => {
+      casesClient = (await endpointAppContextService.getCasesClient(
+        {} as KibanaRequest
+      )) as CasesClientMock;
+
+      let counter = 1;
+      casesClient.cases.getCasesByAlertID.mockImplementation(async () => {
+        return [
+          {
+            id: `case-${counter++}`,
+            title: 'case',
+          },
+        ];
+      });
+    });
+
+    it('logs a comment to the provided case', async () => {
+      await callRoute(ISOLATE_HOST_ROUTE, {
+        body: { agent_ids: ['XYZ'], case_ids: ['one', 'two'] },
+      });
+
+      expect(casesClient.attachments.add).toHaveBeenCalledTimes(2);
```

Comment:
This is probably focus in the wrong place. We don't really care _how_ the isolation handler gets the comments into the cases, just that they do end up there.

Right now, we just had to call it once-per-case because that's what we had available to call. But we don't need to limit ourselves to that approach if that changes in the future (e.g. a bulk-comment function gets added in the future).

The expectations should ideally just make sure that the case IDs we gave it appear _somewhere_ in the N (we don't really care how many) calls to the comment service. 

It might feel like a small distinction, but as-written, this test is pretty easy to break in the future while still providing the correct end behavior.

Reply:
What would you suggest?
The goal here is to test that `case_ids` are used when they are provided on input to the API with no `alert_ids`

Reply:
this isn't currently checking that the right `case_ids` are used. Just that `add` is called twice.

_Twice_ being the part that I think is incorrect to test for. Checking that is it called _at least once_ is great. `.toHaveBeenCalled()` seems fine to accomplish that.

Checking for the right IDs, you should collect up the arguments provided to `casesClient.attachments.add.mock.calls` and make sure that `"one" and "two"` show up, order independent.